app/src/main/java/com/hamdan/forzenbook/data/repository/CreateAccountRepositoryImpl.kt

```diff
@@ -12,14 +15,17 @@ class CreateAccountRepositoryImpl(
         birthDay: String,
         email: String,
         location: String
-    ): Int {
-        return try {
-            service.createUser(firstName, lastName, birthDay, email, location)
-                .code()
-        } catch (e: Exception) {
-            Log.v("Hamdan", "There was an unidentifiable issue when creating the account")
-            0
+    ) {
+        when (
+            service.createUser(
+                email, Date.valueOf(birthDay), firstName, lastName, location
+            ).code()
+        ) {
+            in (SUCCESS) -> {}
```

Comment:
No lol this is sloppy. Especially putting a constant for a range. 
Check isSuccessful, the function from Response for if it's successful. Afterward check codes. Afterward throw

Reply:
ok, I was a little confused on how to apply the isSuccesful with the when block, I hadn't considered just discarding it